src/include/wiredtiger.in

```diff
@@ -5278,6 +5321,286 @@ struct __wt_storage_source {
      */
     int (*terminate)(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session);
 };
+
+/*!
+ * The interface implemented by applications to provide a page log service
+ * implementation.
+ *
+ * <b>Thread safety:</b> WiredTiger may invoke methods on the WT_PAGE_LOG
+ * interface from multiple threads concurrently. It is the responsibility of
+ * the implementation to protect any shared data.
+ *
+ * <b>Reentrancy:</b>Methods on the WT_PAGE_LOG interface are not expected to
+ * support reentrant use.
+ *
+ * Applications register implementations with WiredTiger by calling
+ * WT_CONNECTION::add_page_log.
+ *
+ * @snippet ex_page_log.c WT_PAGE_LOG register
+ */
+struct __wt_page_log {
+
+    /*!
+     * A reference is added to the page log service.  The reference is released by a
+     * call to WT_PAGE_LOG::terminate.  A reference is added as a side effect
+     * of calling WT_CONNECTION::get_page_log.
+     *
+     * @errors
+     *
+     * @param page_log the WT_PAGE_LOG
+     */
+    int (*pl_add_reference)(WT_PAGE_LOG *page_log);
+
+    /*!
+     * Begin checkpointing using the given checkpoint_id.  After this call, any handle
+     * can put or get using the checkpoint id.  The checkpoint id must be greater than
+     * any previous checkpoint id used with this call.
+     *
+     * @errors
+     *
+     * @param page_log the WT_PAGE_LOG
+     * @param session the current WiredTiger session
+     * @param checkpoint_id the checkpoint id to use. Must be greater than any other
+     *        checkpoint_id used with this call.
+     */
+    int (*pl_begin_checkpoint)(WT_PAGE_LOG *page_log, WT_SESSION *session, uint64_t checkpoint_id);
+
+    /*!
+     * Complete checkpointing using the given checkpoint_id. This implies that other
+     * nodes can now use the given checkpoint_id. (Do not use - will be deprecated.)
+     *
+     * @errors
+     *
+     * @param page_log the WT_PAGE_LOG
+     * @param session the current WiredTiger session
+     * @param checkpoint_id the checkpoint id to use. Must be greater than any other
+     *        checkpoint_id used with this call.
+     */
+    int (*pl_complete_checkpoint)(WT_PAGE_LOG *page_log, WT_SESSION *session,
+        uint64_t checkpoint_id);
```

Comment:
There is a notion that this interface shouldn't be used due to deprecation, why should we add it then?

Reply:
Good spot -- thanks. Made WT-14591.